File: 1503-last-moment-before-all-ants-fall-out-of-a-plank/1503-last-moment-before-all-ants-fall-out-of-a-plank.js
/**
 * @param {number} n
 * @param {number[]} left
 * @param {number[]} right
 * @return {number}
 */
var getLastMoment = function(n, left, right) {
    const maxLeft = left.length === 0 ? 0 : Math.max(...left);
    const minRight = right.length === 0 ? n : Math.min(...right);
    return Math.max(maxLeft, n - minRight);
};

File: 1503-last-moment-before-all-ants-fall-out-of-a-plank/1503-last-moment-before-all-ants-fall-out-of-a-plank.java
public class Solution {
    public int getLastMoment(int n, int[] left, int[] right) {
        int lastMoment = 0;
        for(int ant : left){
            lastMoment = Math.max(lastMoment, ant);
        }
        for(int ant : right){
            lastMoment = Math.max(lastMoment, n - ant);
        }
        return lastMoment;
    }
}